### PR TITLE
[TRB-42127]: Introduce PEER_TO_PEER_AFFINITY commodity

### PR DESCRIPTION
**Pull request by andreiastra**
**Purpose:**

Introduce PEER_TO_PEER_AFFINITY commodity. See related:

https://github.ibm.com/turbonomic/opsmgr/pull/194
https://github.ibm.com/turbonomic/api/pull/91
https://github.ibm.com/turbonomic/xl/pull/577

**Testing:**

Compiled and ran Kubeturbo with the new turbo-go-sdk.

### CODE REVIEW

File: pkg/proto/CommonDTO.pb.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.30.0
```
Comment (andreiastra) [inline pkg/proto/CommonDTO.pb.go:3]: @irfanurrehman Should I upgrade it?

File: pkg/proto/CommonDTO.pb.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-// 	protoc        v3.21.12
+// 	protoc-gen-go v1.28.1
```
Comment (andreiastra) [inline pkg/proto/CommonDTO.pb.go:3]: @irfanurrehman Should I upgraded it?

Comment (irfanurrehman) [reply pkg/proto/CommonDTO.pb.go:3]: @andreiastra  you should use the same version of protoc-gen that you see in code to ensure we do not change the autogenerated code by going down the versions.

**Review by irfanurrehman — COMMENTED** [review]
@andreiastra please regenerate this with the right version of protoc-gen

File: pkg/proto/CommonDTO.pb.go [hunk 3]
```diff
@@ -3690,9 +3811,15 @@ const (
...
-	GroupDTO_Setting_MAX_OBSERVATION_PERIOD                GroupDTO_Setting_SettingType = 30
-	GroupDTO_Setting_VCPU_MAX_THROTTLING_TOLERANCE         GroupDTO_Setting_SettingType = 31
+	// The observation period settings (29 and 30) will map to the container spec entity
+	// settings. Since the name of setting type is not specific to container spec, this is
+	// bound to cause confusion. Also, due to backwards compatibility, the name will not be
+	// changed.
+	GroupDTO_Setting_MIN_OBSERVATION_PERIOD        GroupDTO_Setting_SettingType = 29
+	GroupDTO_Setting_MAX_OBSERVATION_PERIOD        GroupDTO_Setting_SettingType = 30
+	GroupDTO_Setting_VCPU_MAX_THROTTLING_TOLERANCE GroupDTO_Setting_SettingType = 31
+	GroupDTO_Setting_MIN_OBSERVATION_PERIOD_VM     GroupDTO_Setting_SettingType = 32
```
Comment (andreiastra) [inline pkg/proto/CommonDTO.pb.go:3821]: Please note that the diff includes somebody else changes.

File: pkg/proto/CommonDTO.pb.go (hunk 2 above)
Comment (andreiastra) [reply pkg/proto/CommonDTO.pb.go]: Upgraded `protoc-gen-go`. The versions look good now.

**Comment by andreiastra** [comment]
Upgraded Go plugin for Protobuf. The versions look good now.

**Review by irfanurrehman — APPROVED** [review]

**Pull request merged**